src/nix/repl.cc

```diff
@@ -514,6 +516,28 @@ bool NixRepl::processLine(string line)
                 logger->cout("  %s -> %s", outputName, state->store->printStorePath(outputPath));
         } else if (command == ":i") {
             runNix("nix-env", {"-i", drvPathRaw});
+        } else if (command == ":log") {
+            settings.readOnlyMode = true;
+            Finally roModeReset([&]() {
+                settings.readOnlyMode = false;
+            });
+            auto subs = getDefaultSubstituters();
+
+            subs.push_front(state->store);
+
+            bool foundLog = false;
+            RunPager pager;
+            for (auto & sub : subs) {
+                auto log = sub->getBuildLog(drvPath);
+                if (log) {
+                    stopProgressBar();
```

Comment:
Hm, why do we need to stop the progress bar?

Reply:
This is done in `nix/log.cc` too, which is where it was copied from. Not doing so seems to leave the progress bar there after the operation is finished.